code/src/components/ThoughtsList.js

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from "react"
+
+import { useEffect } from "react"
+import { ThoughtCard } from "./ThoughtCard"
+
+export const ThoughtsList = () => { 
+    const ThoughtsUrl= "https://happy-thoughts-technigo.herokuapp.com/thoughts";
+    const [thoughts, setThoughts] = useState([]);
+
+    // useEffect prevents the infinite looping of creating new arrays after the data is fetched. 
+    useEffect (() => { 
+      fetchThoughts()
+    }, []);
+
+    const fetchThoughts = () => {
+      fetch(ThoughtsUrl)
+      .then ((response) => {
+          return response.json()
+      })
+      .then (data => { 
+          const filteredThoughts = data.filter(thought => {
+              return thought !== '';
+          });
+          setThoughts(filteredThoughts);     
+      });
+    }
+
+    const onLiked = thoughtId => {
+      const updatedThoughts = thoughts.map(thought => {
+        if (thought._id === thoughtId) {
+          thought.hearts += 1
+        }
+        return thought
+      })
+      setThoughts(updatedThoughts)
+    }
+
+    return (
+        <>
+          <section className="thoughts-container">
```

Comment:
No need for React fragment (`</>`) if you return only one element (in your case `section`)

Reply:
Fixed :)